modules/core/src/main/java/org/apache/synapse/mediators/store/MessageStoreMediator.java

```diff
@@ -78,7 +81,14 @@ public boolean mediate(MessageContext synCtx) {
         }
 
         if(synCtx != null) {
-            MessageStore messageStore = synCtx.getConfiguration().getMessageStore(messageStoreName);
+
+            MessageStore messageStore;
+
+            if(messageStoreExp != null)
```

Comment:
if-else opening and closing brackets are missing.

Reply:
done